packages/legend-query-builder/src/components/shared/BasicValueSpecificationEditor.tsx

```diff
@@ -443,9 +443,14 @@ const setCollectionValue = (
     return;
   }
   let result: unknown[] = [];
-  const parseResult = CSVParser.parse<string[]>(value.trim(), {
+  let parseResult = CSVParser.parse<string[]>(value.trim(), {
     delimiter: ',',
   });
+  if (parseResult.data[0]?.length === 1) {
```

Comment:
we should abstract this out to an utility function and test with some inputs, e.g. `1,3,4,6\n7,8,8,9\n,10`